### PR TITLE
Update agent.rb

### PR DESCRIPTION
**Pull request by Fguhhla**
Add excetion handle to read_paket because of the following error while using net-ssh 6.1 with test-kitchen 

from /opt/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/net-ssh-6.1.0/lib/net/ssh/authentication/agent.rb:211:in read_packet' /opt/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/net-ssh-6.1.0/lib/net/ssh/buffer.rb:147:in append': can't modify frozen String: "" (FrozenError)

### CODE REVIEW

**Comment by mfazekas** [comment]
@Fguhhla thanks much for the PR, I'd be more interested in the reason for the crash vs such a workaround, first it silences errors, second it prints to stdout which might not be appropriate for all apps.

**Comment by mfazekas** [comment]
Isn't it possible that the code was ran In all strings frozen mode?! 
RUBYOPT="--enable-frozen-string-literal"

File: lib/net/ssh/authentication/agent.rb [hunk 1]
```diff
@@ -222,7 +222,12 @@ def send_packet(type, *args)
         # is returned as a Net::SSH::Buffer).
         def read_packet
           buffer = Net::SSH::Buffer.new(@socket.read(4))
-          buffer.append(@socket.read(buffer.read_long))
+          begin
+            buffer.append(@socket.read(buffer.read_long))
+          rescue StandardError => e  
+            puts e.message
```
Comment (fwininger) [inline lib/net/ssh/authentication/agent.rb:228]: Ignore StandardError here is a very bad choice.

**Review by fwininger — CHANGES_REQUESTED** [review]

**Comment by tas50** [comment]
@mfazekas We freeze strings throughout Test Kitchen

**Comment by mfazekas** [comment]
@tas50 can you elaborate?! Do you use `enable-frozen-string-literal` globally ?!  `net-ssh` doesn't support this at the moment. [Buffer](https://github.com/net-ssh/net-ssh/blob/master/lib/net/ssh/buffer.rb) for sure is not frozen literal compatible.

**Comment by tas50** [comment]
You know it looks like we're not actually doing it in 100% of the files like I thought. We do set the frozen string literal magic comment in a large number of the Test Kitchen files though since it has a nice memory impact and at one point the world was prepping for that as the default in Ruby 3.

**Comment by mfazekas** [comment]
@Fguhhla thanks much for the PR. We'd need to understand why this error happens, just silencing it is not solution. 

I'm closing this PR, let us know if you have more background info on this.

**Pull request closed without merging**